src/com/xmslyz/luhn/logic/Number.java

```diff
@@ -0,0 +1,33 @@
+package com.xmslyz.luhn.logic;
+
+class Number {
+    private int length;
+
+    public Number(int length) {
+        setLength(length);
+    }
+
+//    public Number(int length) {
+//        //this.type = type;
+//        this.length = length;
+//        //this.itsClass = itsClass;
+//    }
+
+//    public String getType() {
+//        String type = "digit";
+//        return type;
+//    }
+
+//    public String getItsClass() {
+//        String itsClass = "String";
+//        return itsClass;
+//    }
+
+    public int getLength() {
+        return length;
+    }
+
+    public void setLength(int length) {
+        this.length = length;
+    }
+}
```

Comment:
zamiast całego tego kodu możesz użyć [record](https://docs.oracle.com/en/java/javase/14/language/records.html):
```
record Number(int length) {}
```
Ale po prawdzie w ogóle nie potrzebujesz tej klasy. Jedyne jej użycie wygląda tak:
```
        Number numberToCodeLenght = new Number(number.length());
        return numberToCodeLenght.getLength();
```
a możesz to przecież skrócić do:
```
        return number.length();
```

Reply:
Jeśli wiesz dokładnie jak cos będzie zaimplementowane, to można zostawić na to takie miejsce. Ale gdy nie masz pewności jak i wiesz tylko że "jakoś", to lepiej nie dodawać od razu kodu, który kiedyś się przyda.  

Przy tworzeniu softu, warto sobie wziąć do serca trzy reguły: [KISS, DRY oraz YAGNI](https://www.boldare.com/blog/kiss-yagni-dry-principles/)